platforms/documentation/docs/src/docs/userguide/dep-man/declaring-depedencies/declaring_configurations.adoc

```diff
@@ -0,0 +1,148 @@
+== Understanding configuration inheritance
+
+Configurations can inherit from other configurations, creating an inheritance hierarchy.
+Child configurations inherit all dependencies declared in their parent configurations.
+
+Configuration inheritance is commonly used by Gradle core plugins, such as the <<java_plugin.adoc#sec:java_plugin_and_dependency_management,Java plugin>>.
+For instance, the `testImplementation` configuration extends the `implementation` configuration.
+
+image::dependency-management-configuration-inheritance.png[]
+
+This hierarchy ensures that dependencies needed for both compiling tests and the source code under test are included.
+For example, a Java project using JUnit for tests might also need Guava if it's used in the production code.
+
+Internally, configurations form an inheritance hierarchy using the link:{groovyDslPath}/org.gradle.api.artifacts.Configuration.html#org.gradle.api.artifacts.Configuration:extendsFrom(org.gradle.api.artifacts.Configuration++[]++)[`Configuration.extendsFrom(Configuration...)`] method.
+A configuration can extend any other configuration, regardless of how it is defined in the build script or plugin.
+
+For instance, if you want to create a suite of smoke tests that verify a web service endpoint using JUnit, you can define a new configuration named `smokeTest` that extends from `testImplementation` to reuse the existing test framework dependency:
+
+====
+include::sample[dir="snippets/dependencyManagement/definingUsingConfigurations-inheritance/kotlin",files="build.gradle.kts[tags=configuration-definition]"]
+include::sample[dir="snippets/dependencyManagement/definingUsingConfigurations-inheritance/groovy",files="build.gradle[tags=configuration-definition]"]
+====
+
+NOTE: Configurations can only extend configurations within the same project.
+
+[[sec:defining-custom-configurations]]
+== Creating custom configurations
+
+You can define custom configurations to separate the scope of dependencies for specific purposes.
+
+For example, to use the link:https://tomcat.apache.org/tomcat-9.0-doc/jasper-howto.html[Jasper Ant task] for pre-compiling JSP files without including it in the classpath for compiling your source code, you can create and use a custom configuration:
+
+====
+include::sample[dir="snippets/dependencyManagement/definingUsingConfigurations-custom/kotlin",files="build.gradle.kts[tags=custom-configuration]"]
+include::sample[dir="snippets/dependencyManagement/definingUsingConfigurations-custom/groovy",files="build.gradle[tags=custom-configuration]"]
+====
+
+You can manage custom configurations using the `configurations` object.
+Configurations can have names and can extend each other.
+For more details, refer to the link:{groovyDslPath}/org.gradle.api.artifacts.ConfigurationContainer.html[`ConfigurationContainer`] API.
+
+Remember that configurations are intended to be used for a single role: declaring dependencies, performing resolution, or defining consumable variants.
```

Comment:
You say "remember" here, but has this information been described anywhere before? 

I think we should link to that "prior knowledge" if we're going to use language like this

Reply:
This is a leftover from the old docs and has now been rewritten.